src/components/Modal/Modal.stories.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from "react";
+import { Meta, StoryFn } from "@storybook/react";
+import { BrowserRouter } from "react-router-dom";
+import { Modal } from "./Modal";
+
+export default {
+  title: "components/Modal",
+  component: Modal,
+  argTypes: {
+    isOpen: {
+      control: "boolean",
+      description: "Toggles the modal open and close state",
+    },
+  },
+} as Meta;
+
+const Template: StoryFn<{ isOpen: boolean }> = ({ isOpen }) => {
+  const [modalOpen, setModalOpen] = useState(isOpen);
+
+  // Effect to sync Storybook control with local state
+  React.useEffect(() => {
+    setModalOpen(isOpen);
+  }, [isOpen]);
+
+  return (
+    <BrowserRouter>
+      <Modal
+        name="modal"
+        isOpen={modalOpen}
+        onClose={() => setModalOpen(false)} // Close modal on overlay click
```

Comment:
When using the storybook here, if I click outside the Modal, it does close, but the state variable `isOpen` stays at `true`. It also doesn't fade or animate closed. Can we take a look at why that's happening? I also agree with @raymondafan 's comment below about potentially adding a button in the storybook to open the modal in the first place.